app/views/rails_admin/main/index.html.haml

```diff
@@ -12,10 +12,22 @@
   properties = @model_config.list.with(controller: self.controller, view: self, object: @abstract_model.model.new).visible_fields
   checkboxes = @model_config.list.checkboxes?
   # columns paginate
-  sets = get_column_sets(properties)
-  properties = sets[params[:set].to_i] || []
-  other_left = ((params[:set].to_i - 1) >= 0) && sets[params[:set].to_i - 1].present?
-  other_right = sets[params[:set].to_i + 1].present?
+  horiz_scroll_config = RailsAdmin::Config.horizontal_scroll_list
+  horiz_scroll_enabled = !!horiz_scroll_config
+  if horiz_scroll_enabled
+    if horiz_scroll_config.is_a?(Hash)
+      horiz_scroll_frozen_cols = horiz_scroll_config[:num_frozen_columns]
+      horiz_scroll_css = horiz_scroll_config[:css]
+    end
+    horiz_scroll_frozen_cols ||= 3 # by default, freeze checkboxes, links & first property (usually id?)
+    horiz_scroll_frozen_cols -= 1 unless checkboxes
+  else
+    horiz_scroll_frozen_cols = 0
+    sets = get_column_sets(properties)
+    properties = sets[params[:set].to_i] || []
+    other_left = ((params[:set].to_i - 1) >= 0) && sets[params[:set].to_i - 1].present?
+    other_right = sets[params[:set].to_i + 1].present?
+  end
```

Comment:
Can these calculation logics be more simple?
If they needs to be that complex, consider moving it to outside of the view.